### ISSUE TITLE
Couldn't find template for digesting: Error when calling a dynamically named partial

### ISSUE THREAD

**Issue by lustremedia** [rails/rails#19198]
I am having an issues when trying to cache dynamically named partials which include static content

In site.html.erb I call:
```
<% cache("#{@site.name}_#{locale}", :expires_in => 2.hour) do %>
    <%= render partial: "my_#{@site_name}_#{locale}" %>
<% end %>
```
Error
```
Couldn't find template for digesting: pages/my_#{@site_name}_#{locale}
Cache digest for app/views/pages/site.html.erb: 493b4a427278b191dc524264e105ea52
```
The partial is located in the same directory as ```site.html.erb``` and is e.g. named as follows: ```pages/_my_nice_page_en.html.erb``` or ```pages/_my_nice_page_de.html.erb```

It looks like if rails looks at the wrong place for partials and or does not evaluate the dynamic string assignment and thus can't assign a digest to a missing partial and thus cannot cache the partials.

**Comment by rafaelfranca** [comment]
Yeah, it shot not try to calculate digest for this case. There is no way to evaluate the dynamic string.

**Comment by lustremedia** [comment]
So what would be the workaround? Move the caching inside the partials?

**Comment by rafaelfranca** [comment]
Actually there is a problem on Rails, since it should not try to calculate the digest for the partial.

Does it give an error for the users or just in the log?

**Comment by lustremedia** [comment]
No user error! it rescues in https://github.com/rails/rails/blob/08754f12e65a9ec79633a605e986d0f1ffa4b251/actionview/lib/action_view/digestor.rb#66 an that is the error I am getting.

**Comment by lustremedia** [comment]
btw I am on Rails 4.1.7

**Comment by rafaelfranca** [comment]
If there is no user error you don't need any workaround. The code will continue just ignoring this template, what is correct.

**Comment by bogdan** [comment]
@rafaelfranca so do you think it is correct to generate that error message in log?

**Comment by rafaelfranca** [comment]
Yes, it is correct. The digest generation should not make the code stop to work. And there is no fix for this bug unless making Rails to stop to calculate the digest since we can't evaluate the dynamic string.

**Comment by PeterBengtson** [comment]
Will someone fix this? It's still in 4.2.4 and it's still messing up the logs.

**Comment by sevenseacat** [comment]
@PeterBengtson are you volunteering? :)

**Comment by PeterBengtson** [comment]
Absolutely not. I can live with it, I just need to code around it. Fixing it is much more efficiently done by someone who has worked with that particular subsystem. And it is a core bug.

**Comment by chrishough** [comment]
Has anyone found an fix or work around for this? It seems like it is happening in other places now too, not just partials.

**Comment by justinperkins** [comment]
Would be nice to be able to exclude a partial from this digest builder, rather than just be trained to ignore the log messages. Not to mention bypassing the digest tap+rescue block.

**Comment by beder** [comment]
Just for the record, it looks like this issue was solved here: bb048147912d7ea8c301acd6468e9f869dd0acad.

**Comment by vemv** [comment]
Is there a known-good approach to dynamically render different partials and having those cached?

While the curreny behavior isn't buggy and the log 'noise' doesn't annoy me much, the actual application performance suffers.
